Can't play ultrawide content on Samsung OLED G8

Like many others, the JellyFin Tizen app does not run in the native resolution of 3440 x 1440 pixels but in a 16:9 box, centered on the screen.
21:9 content gets displayed with 4 black bars or if cropped, displays black bars on the left and right instead of the movie.
I tried putting my resolution in the tizen.js file and some other places but that did not fix the issue.

The Netflix and Prime Video apps also behave like this so I asked the Netflix support about this and they confirmed the issue. "The experts will look into it." They also keep pointing fingers at Samsung because "They have to make sure that first party apps run properly on their devices."

I don't know who is to blame in the case of Netflix but displaying apps in the full resolution has to be possible since the YouTube app does just that. No cropping needed, it displays ultra wide content without any black bars.

![Image](https://github.com/user-attachments/assets/9222bffd-ea51-4b94-a9f6-de4072da917e)

![Image](https://github.com/user-attachments/assets/aa2b7985-5796-4de2-b629-d645f668bb0c)

After changing these two files I get an NPM error 1,
```

user@DESKTOP-I74KDL4:~$ mkdir /mnt/c/Jellyfin/
user@DESKTOP-I74KDL4:~$ cd /mnt/c/Jellyfin/
user@DESKTOP-I74KDL4:/mnt/c/Jellyfin$ git clone -b release-10.10.z https://github.com/jellyfin/jellyfin-web.git
Cloning into 'jellyfin-web'...
remote: Enumerating objects: 245280, done.
remote: Counting objects: 100% (100/100), done.
remote: Compressing objects: 100% (49/49), done.
remote: Total 245280 (delta 80), reused 55 (delta 51), pack-reused 245180 (from 4)
Receiving objects: 100% (245280/245280), 170.29 MiB | 6.41 MiB/s, done.
Resolving deltas: 100% (171932/171932), done.
Updating files: 100% (1040/1040), done.
user@DESKTOP-I74KDL4:/mnt/c/Jellyfin$ git clone https://github.com/jellyfin/jellyfin-tizen.git
Cloning into 'jellyfin-tizen'...
remote: Enumerating objects: 404, done.
remote: Counting objects: 100% (160/160), done.
remote: Compressing objects: 100% (95/95), done.
remote: Total 404 (delta 99), reused 65 (delta 65), pack-reused 244 (from 2)
Receiving objects: 100% (404/404), 286.16 KiB | 2.29 MiB/s, done.
Resolving deltas: 100% (202/202), done.
user@DESKTOP-I74KDL4:/mnt/c/Jellyfin$ cd jellyfin-web
npm ci --no-audit
USE_SYSTEM_FONTS=1 npm run build:production
npm warn deprecated @types/localforage@0.0.34: This is a stub types definition for localforage (https://github.com/localForage/localForage). localforage provides its own type definitions, so you don't need @types/localforage installed!
npm warn deprecated @types/vfile-message@2.0.0: This is a stub types definition. vfile-message provides its own type definitions, so you do not need this installed.
npm warn deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm warn deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm warn deprecated @stylelint/postcss-markdown@0.36.2: Use the original unforked package instead: postcss-markdown
npm warn deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm warn deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm warn deprecated @stylelint/postcss-css-in-js@0.37.3: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.

added 1873 packages in 6m

> jellyfin-web@10.10.3 build:production
> cross-env NODE_ENV="production" webpack --config webpack.prod.js

[BABEL] Note: The code generator has deoptimised the styling of /mnt/c/Jellyfin/jellyfin-web/node_modules/pdfjs-dist/build/pdf.js as it exceeds the max of 500KB.
user@DESKTOP-I74KDL4:/mnt/c/Jellyfin/jellyfin-web$ cd -
cd -
/mnt/c/Jellyfin
user@DESKTOP-I74KDL4:/mnt/c/Jellyfin$ cd jellyfin-tizen
user@DESKTOP-I74KDL4:/mnt/c/Jellyfin/jellyfin-tizen$ JELLYFIN_WEB_DIR=../jellyfin-web/dist npm ci --no-audit
npm warn deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm warn deprecated har-validator@5.1.5: this library is no longer supported
npm warn deprecated domexception@1.0.1: Use your platform's native DOMException instead
npm warn deprecated abab@2.0.5: Use your platform's native atob() and btoa() methods instead
npm warn deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm warn deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

> jellyfin-tizen@0.1.0 postinstall
> gulp

[02:24:29] Loaded external module: @babel/register
Error: Cannot find module '@babel/preset-env'
Require stack:
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/plugins.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/index.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/index.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker/babel-core.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker/handle-message.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker-client.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/node.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/nodeWrapper.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/index.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/rechoir/lib/register.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/rechoir/index.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/liftoff/lib/register_loader.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/liftoff/index.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/gulp-cli/index.js
- /mnt/c/Jellyfin/jellyfin-tizen/node_modules/gulp/bin/gulp.js

Make sure that all the Babel plugins and presets you are using
are defined as dependencies or devDependencies in your package.json
file. It's possible that the missing plugin is loaded by a preset
you are using that forgot to add the plugin to its dependencies: you
can workaround this problem by explicitly adding the missing package
to your top-level package.json.

    at Function._resolveFilename (node:internal/modules/cjs/loader:1249:15)
    at resolve (node:internal/modules/helpers:148:19)
    at tryRequireResolve (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/plugins.js:121:11)
    at resolveStandardizedNameForRequire (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/plugins.js:155:19)
    at resolveStandardizedName (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/plugins.js:170:12)
    at loadPreset (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/plugins.js:61:20)
    at loadPreset.next (<anonymous>)
    at createDescriptor (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/config-descriptors.js:140:16)
    at createDescriptor.next (<anonymous>)
    at evaluateSync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:251:28)
    at /mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:31:34
    at Array.map (<anonymous>)
    at Function.sync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:31:22)
    at Function.all (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:210:24)
    at Generator.next (<anonymous>)
    at createDescriptors (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/config-descriptors.js:102:41)
    at createDescriptors.next (<anonymous>)
    at createPresetDescriptors (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/config-descriptors.js:96:17)
    at createPresetDescriptors.next (<anonymous>)
    at /mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/config-descriptors.js:57:32
    at Generator.next (<anonymous>)
    at Function.<anonymous> (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/gensync-utils/async.js:21:3)
    at Generator.next (<anonymous>)
    at evaluateSync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:251:28)
    at Function.sync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:89:14)
    at sync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/gensync-utils/async.js:61:25)
    at sync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:182:19)
    at onFirstPause (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:210:24)
    at Generator.next (<anonymous>)
    at cachedFunction (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/caching.js:52:46)
    at cachedFunction.next (<anonymous>)
    at mergeChainOpts (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/config-chain.js:350:34)
    at mergeChainOpts.next (<anonymous>)
    at chainWalker (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/config-chain.js:316:14)
    at chainWalker.next (<anonymous>)
    at buildRootChain (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/config-chain.js:56:36)
    at buildRootChain.next (<anonymous>)
    at loadPrivatePartialConfig (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/partial.js:72:62)
    at loadPrivatePartialConfig.next (<anonymous>)
    at loadFullConfig (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/full.js:36:46)
    at loadFullConfig.next (<anonymous>)
    at loadOptionsImpl (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/index.js:52:43)
    at loadOptionsImpl.next (<anonymous>)
    at evaluateSync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:251:28)
    at sync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gensync/index.js:89:14)
    at stopHiding - secret - don't use this - v1 (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/errors/rewrite-stack-trace.js:47:12)
    at loadOptionsSync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/index.js:60:78)
    at OptionManager.init (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/index.js:233:42)
    at exports.transformSync (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker/transform.js:67:44)
    at LocalClient.handleMessage (/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker/handle-message.js:18:16) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/plugins.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/config/files/index.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/core/lib/index.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker/babel-core.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker/handle-message.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/worker-client.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/node.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/nodeWrapper.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/@babel/register/lib/index.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/rechoir/lib/register.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/rechoir/index.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/liftoff/lib/register_loader.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/liftoff/index.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gulp-cli/index.js',
    '/mnt/c/Jellyfin/jellyfin-tizen/node_modules/gulp/bin/gulp.js'
  ]
}
npm error code 1
npm error path /mnt/c/Jellyfin/jellyfin-tizen
npm error command failed
npm error command sh -c gulp
npm error A complete log of this run can be found in: /home/user/.npm/_logs/2024-12-19T01_23_51_386Z-debug-0.log
user@DESKTOP-I74KDL4:/mnt/c/Jellyfin/jellyfin-tizen$
```

--- a/config.xml
+++ b/config.xml
@@ -7,6 +7,7 @@
     <description>Jellyfin for Samsung Smart TV (Tizen).</description>
     <feature name="http://tizen.org/feature/screen.size.all"/>
     <icon src="icon.png"/>
+    <tizen:metadata key="http://tizen.org/metadata/app_ui_type/base_screen_resolution" value="fullscreen"/>
     <name>Jellyfin</name>
     <tizen:privilege name="http://developer.samsung.com/privilege/productinfo"/>
     <tizen:privilege name="http://tizen.org/privilege/tv.inputdevice"/>

